pkg/devicerepository/grpc.go

```diff
@@ -172,32 +172,29 @@ func (dr *DeviceRepository) GetTemplate(ctx context.Context, req *ttnpb.GetTempl
 	return dr.store.GetTemplate(req.VersionIDs)
 }
 
-// GetUplinkDecoder implements the ttnpb.DeviceRepositoryServer interface.
-func (dr *DeviceRepository) GetUplinkDecoder(ctx context.Context, req *ttnpb.GetPayloadFormatterRequest) (*ttnpb.MessagePayloadFormatter, error) {
+func getCodec(ctx context.Context, req *ttnpb.GetPayloadFormatterRequest, f func(store.GetCodecRequest) (*ttnpb.MessagePayloadFormatter, error)) (*ttnpb.MessagePayloadFormatter, error) {
 	if clusterauth.Authorized(ctx) != nil {
 		if err := rights.RequireApplication(ctx, req.ApplicationIDs, ttnpb.RIGHT_APPLICATION_DEVICES_READ); err != nil {
 			return nil, err
 		}
 	}
-	return dr.store.GetUplinkDecoder(req.VersionIDs)
+	return f(store.GetCodecRequest{
+		VersionIDs: req.VersionIDs,
+		Paths:      ttnpb.AddFields(req.FieldMask.Paths, "formatter", "formatter_parameter"),
```

Comment:
This ensures API backwards compatibility, which means that it is OK even if this does not come with 3.12.0.

Moving to 3.12.1 milestone.